### PR TITLE
FIX: Only pass volumetric spaces to carpetplot

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

We had a situation where `['fsaverage', 'MNI152NLin2009c']` was being passed as the set of spaces to `carpetplot_wf`, but only the transforms for MNI were being passed, so the indexing was getting screwed up. It appears that the resolution is simply to pass the volumetric spaces.

This resolves the specific failure documented in https://github.com/poldracklab/fmriprep/issues/1921#issuecomment-572206926.

### CODE REVIEW

**Comment by effigies** [comment]
@mgxd @oesteban Any concerns?

File: fmriprep/workflows/bold/base.py [hunk 1]
```diff
@@ -817,7 +825,7 @@ def init_func_preproc_wf(
 
         if 'MNI152NLin2009cAsym' in std_spaces:
             carpetplot_wf = init_carpetplot_wf(
-                standard_spaces=std_spaces,
+                standard_spaces=volume_std_spaces,
```
Comment (oesteban) [inline fmriprep/workflows/bold/base.py]: I believe this line needs to go away, and pass the list of spaces through a connection to the collated list of templates.
```suggestion
```
That means adding a line right after:
https://github.com/poldracklab/fmriprep/pull/1932/files#diff-e705526667f071ca0c3b04c2bd96272bR834

Like
```Python
                    ('joint_std2anat_xfm', 'inputnode.std2anat_xfm'),
                    ('joint_template', 'inputnode.templates')]),
```

Unfortunately, the carpetplot workflow would not support this directly:

- we need to remove the function argument
- we need to add a new inputnode field
- we need to then connect the new field to `select_std.inputs.keys`

**Comment by effigies** [comment]
Bearing in mind that this is a bug-fix, and the final fix will be #1915, is there a use-case that your suggestion fixes that this doesn't? I would like to keep this as small as possible to reduce the potential for new bugs.

Will wait on response before diving into effigies#2.

**Comment by effigies** [comment]
Merged effigies#2. Needed small fixes that I added.

@mgxd If we're replacing `KeySelect` anyway, this just moves it from one subworkflow to another. Apologies if it causes merge conflicts.

Any objections to merging on passing tests?

**Comment by mgxd** [comment]
Nope

**Pull request merged**